src/mscorlib/src/System/Collections/Generic/List.cs

```diff
@@ -247,7 +246,7 @@ public void Add(T item)
             var array = _items;
             var size = _size;
             _version++;
-            if ((uint)size < (uint)array.Length)
+            if (array.IndexInRange(size))
```

Comment:
In the List`1.Add(long) dasm I was looking at, it never put the length in a register https://github.com/dotnet/coreclr/issues/9707#issue-209303637

```asm
; Assembly listing for method List`1:Add(long):this
; Emitting BLENDED_CODE for X64 CPU with SSE2
; optimized code
; rsp based frame
; fully interruptible
; Final local variable assignments
;
;  V00 this         [V00,T00] ( 12,  10.5)     ref  ->  rcx         this
;  V01 arg1         [V01,T02] (  4,   3  )    long  ->  rdx        
;  V02 loc0         [V02,T03] (  4,   3  )     ref  ->  rax        
;  V03 loc1         [V03,T01] (  7,   5  )     int  ->   r8        
;  V04 OutArgs      [V04    ] (  1,   1  )  lclBlk (32) [rsp+0x00]  
;
; Lcl frame size = 40

G_M46198_IG01:
       4883EC28             sub      rsp, 40
       90                   nop      

G_M46198_IG02:
       488B4108             mov      rax, gword ptr [rcx+8]
       448B4118             mov      r8d, dword ptr [rcx+24]
       FF411C               inc      dword ptr [rcx+28]
       44394008             cmp      dword ptr [rax+8], r8d  // check
       761C                 jbe      SHORT G_M46198_IG04     // jump alt path
       458D4801             lea      r9d, [r8+1]
       44894918             mov      dword ptr [rcx+24], r9d
       443B4008             cmp      r8d, dword ptr [rax+8]  // check
       731C                 jae      SHORT G_M46198_IG06     // jump bounds check fail (unreachable)
       4963C8               movsxd   rcx, r8d
       488954C810           mov      qword ptr [rax+8*rcx+16], rdx

G_M46198_IG03:
       4883C428             add      rsp, 40
       C3                   ret      

G_M46198_IG04:
       488D0500000000       lea      rax, [(reloc)]

G_M46198_IG05:
       4883C428             add      rsp, 40
       48FFE0               rex.jmp  rax

G_M46198_IG06:
       E800000000           call     CORINFO_HELP_RNGCHKFAIL
       CC                   int3     

; Total bytes of code 69, prolog size 5 for method List`1:Add(long):this
```
Though it should get picked up by your change https://github.com/dotnet/coreclr/pull/9773, and its not too onerous to do unsigned casts in the rare cases its needed (or to actually use unsigned indexers in the first place) but it doesn't look as pretty as not using them.